src/algorithm/dft.rs

```diff
@@ -45,6 +45,15 @@ impl<T: FftNum> Dft<T> {
         0
     }
 
+    fn perform_fft_immut(
+        &self,
+        signal: &[Complex<T>],
+        spectrum: &mut [Complex<T>],
+        _scratch: &mut [Complex<T>],
+    ) {
+        self.perform_fft_out_of_place(signal, spectrum, _scratch);
```

Comment:
Minor feedback: For consistency's sake, I'd argue that we should swap these, so that the out of place version calls the immutable version.